satpy/readers/yaml_reader.py

```diff
@@ -1221,25 +1207,94 @@ def _load_dataset(dsid, ds_info, file_handlers, dim='y', pad_data=True):
+        expected_segments = file_handlers[0].filetype_info['expected_segments']
+        filetype = file_handlers[0].filetype_info['file_type']
+        available_segments = [int(fh.filename_info.get('segment', 1)) for
+                              fh in file_handlers]
+        area_defs = {}
 
-    return area_def
+        for segment in range(available_segments[0], expected_segments + 1):
+            try:
+                idx = available_segments.index(segment)
+                fh = file_handlers[idx]
+                area = fh.get_area_def(dsid)
+            except ValueError:
+                logger.debug("Padding to full disk with segment nr. %d", segment)
+
+                new_height_proj_coord, new_height_px = self._get_new_areadef_heights(area, seg_size,
+                                                                                     segment_n=segment,
+                                                                                     filetype=filetype)
+                new_ll_y = area.area_extent[1] + new_height_proj_coord
+                new_ur_y = area.area_extent[1]
+                fill_extent = (area.area_extent[0], new_ll_y,
+                               area.area_extent[2], new_ur_y)
+                area = AreaDefinition('fill', 'fill', 'fill', area.crs,
+                                      seg_size[1], new_height_px,
+                                      fill_extent)
+
+            area_defs[segment] = area
+            seg_size = area.shape
+
+        return area_defs
+
+    def _pad_earlier_segments_area(self, file_handlers, dsid, area_defs):
+        """Pad area definitions for missing segments that are earlier in sequence than the first available."""
+        available_segments = [int(fh.filename_info.get('segment', 1)) for
+                              fh in file_handlers]
+        area = file_handlers[0].get_area_def(dsid)
+        seg_size = area.shape
+        filetype = file_handlers[0].filetype_info['file_type']
+
+        for segment in range(available_segments[0] - 1, 0, -1):
+            logger.debug("Padding segment %d to full disk.",
+                         segment)
+
+            new_height_proj_coord, new_height_px = self._get_new_areadef_heights(area, seg_size,
+                                                                                 segment_n=segment,
+                                                                                 filetype=filetype)
+            new_ll_y = area.area_extent[3]
+            new_ur_y = area.area_extent[3] - new_height_proj_coord
+            fill_extent = (area.area_extent[0], new_ll_y,
+                           area.area_extent[2], new_ur_y)
+            area = AreaDefinition('fill', 'fill', 'fill',
+                                  area.crs,
+                                  seg_size[1], new_height_px,
+                                  fill_extent)
+            area_defs[segment] = area
+            seg_size = area.shape
```

Comment:
This structure is very similar to the one in the exception block in `_pad_later_segments_area`, maybe these could be combined to one function?

Reply:
Done